src/onnx/parse_qlinearconv.cpp

```diff
@@ -0,0 +1,277 @@
+        auto sh_w   = in_w->get_shape();
+        auto type_x = sh_x.type();
+        auto type_w = sh_w.type();
+
+        if(type_x != shape::int8_type and type_x != shape::uint8_type)
+            MIGRAPHX_THROW("QLINEARCONV: unsupported input type");
+        if(type_w != shape::int8_type and type_w != shape::uint8_type)
+            MIGRAPHX_THROW("QLINEARCONV: unsupported weight type");
+        if(in_scale_x->get_shape().type() != shape::float_type)
+            MIGRAPHX_THROW("QLINEARCONV x scale type should be float");
+        if(in_scale_w->get_shape().type() != shape::float_type)
+            MIGRAPHX_THROW("QLINEARCONV: wt scale type should be float");
+        if(in_scale_y->get_shape().type() != shape::float_type)
+            MIGRAPHX_THROW("QLINEARCONV: y scale type should be float");
+        if(inp_arg.size() > 8 and inp_arg[8]->get_shape().type() != shape::int32_type)
+            MIGRAPHX_THROW("QLINEARCONV y bias should be int32");
+    }
+
+    // process all attributes of QLinearConv Operator..
+    value process_attributes(const onnx_parser& parser,
+                             const onnx_parser::node_info& info,
+                             const std::vector<instruction_ref>& args) const
+    {
+        value values;
+        const auto& in_x = args[0];
+        auto in_x_shape  = in_x->get_shape();
+        auto in_x_lens   = in_x_shape.max_lens();
+        assert(in_x_lens.size() > 2);
+        auto kdims = in_x_lens.size() - 2;
+
+        check_padding_mode(info, "QLINEARCONV");
+
+        if(contains(info.attributes, "strides"))
+        {
+            copy(info.attributes.at("strides").ints(), std::back_inserter(values["stride"]));
+            check_attr_sizes(kdims, values["stride"].size(), "QLINEARCONV: inconsistent strides");
+        }
+
+        if(contains(info.attributes, "dilations"))
+        {
+            copy(info.attributes.at("dilations").ints(), std::back_inserter(values["dilation"]));
+            check_attr_sizes(
+                kdims, values["dilation"].size(), "QLINEARCONV: inconsistent dilations");
+        }
+
+        if(contains(info.attributes, "pads"))
+        {
+            std::vector<int64_t> padding;
+            copy(info.attributes.at("pads").ints(), std::back_inserter(padding));
+            check_attr_sizes(kdims, padding.size() / 2, "QLINEARCONV: inconsistent padding");
+            values["padding"] = std::vector<size_t>(padding.begin(), padding.end());
+        }
+
+        if(contains(info.attributes, "auto_pad"))
+        {
+            auto auto_pad = info.attributes.at("auto_pad").s();
+            if(auto_pad.find("NOTSET") == std::string::npos)
+                MIGRAPHX_THROW("QLINEARCONV: auto_pad option fix..");
+        }
+
+        if(contains(info.attributes, "group"))
+            values["group"] = parser.parse_value(info.attributes.at("group")).template at<int>();
+
+        if(not values.empty())
+            recalc_conv_attributes(values, kdims);
```

Comment:
I don't know why this would be necessary. Could remnant of older logic.

Reply:
Setting one set of attributes can mean checking on other attributes -- which have to change in sync..; (I don't know if it is an older logic.) Thanks.